securedrop/journalist.py

```diff
@@ -593,11 +602,25 @@ def reply():
                         [crypto_util.getkey(g.sid), config.JOURNALIST_KEY],
                         output=store.path(g.sid, filename))
     reply = Reply(g.user, g.source, filename)
-    db_session.add(reply)
-    db_session.commit()
 
-    flash("Thanks! Your reply has been stored.", "notification")
-    return redirect(url_for('col', sid=g.sid))
+    try:
+        db_session.add(reply)
+        db_session.commit()
+    except Exception as exc:
+        flash("An unexpected error occurred! Please check the application "
+              "logs or inform your adminstrator.", "error")
+        # We take a cautious approach to logging here because we're dealing
+        # with responses to sources. It's possible the exception message could
+        # contain information we don't want to write to disk.
+        app.logger.error(
+            "Reply from '{}' (id {}) failed: {}!".format(g.user.username,
+                                                         g.user.id,
+                                                         exc.__class__))
+        db_session.rollback()
```

Comment:
@fowlslegs I don't think you need to call `.rollback()` explicitly. If an exception is raised before or during `.commit()`, then the transaction will have completed, and `.rollback()` will be called automatically when the scoped_session is removed in `shutdown_session`. For details, see: http://flask.pocoo.org/docs/0.12/patterns/sqlalchemy/ and http://docs.sqlalchemy.org/en/latest/orm/contextual.html.